pkg/cloudprovider/cloudprovider.go

```diff
@@ -112,6 +115,59 @@ func withInspection() autorest.PrepareDecorator {
 	}
 }
 
+// GetUserMSIs will return a list of all identities on the node
+func (c *Client) GetUserMSIs(node *corev1.Node) ([]string, error) {
+	idH, _, err := c.getIdentityResource(node)
+	if err != nil {
+		glog.Errorf("GetUserMSIs: get identity resource failed with error %v", err)
+		return nil, err
+	}
+	info := idH.IdentityInfo()
+	if info == nil {
+		return nil, fmt.Errorf("identity info is nil")
+	}
+	idList := info.GetUserIdentityList()
+	return idList, nil
+}
+
+// AppendOrRemoveUserMSI will batch process the removal and addition of ids
+func (c *Client) AppendOrRemoveUserMSI(addUserAssignedMSIIDs []string, removeUserAssignedMSIIDs []string, node *corev1.Node) error {
```

Comment:
Is it possible to split this into two separate functions, one which adds and another which removes and then we can add individually add UTs for them ?

Reply:
Or may be naming can be UpdateUserMSI - AppendOrRemove feels like CreateOrUpdate which is doing a different thing.

Reply:
Renamed the method